src/Features/Core/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -445,27 +445,55 @@ public bool IsAnalyzerSuppressed(DiagnosticAnalyzer analyzer)
 
-                // Get all the analyzer actions, including the per-compilation actions.
-                var analyzerExecutor = GetAnalyzerExecutor(analyzer, compilation, localDiagnostics.Add);
-                var analyzerActions = await this.GetAnalyzerActionsAsync(analyzer, analyzerExecutor).ConfigureAwait(false);
-                var hasDependentCompilationEndAction = await AnalyzerManager.Instance.GetAnalyzerHasDependentCompilationEndAsync(analyzer, analyzerExecutor).ConfigureAwait(false);
-
-                if (hasDependentCompilationEndAction && forceAnalyzeAllDocuments != null)
+                try
                 {
-                    // Analyzer registered a compilation end action and at least one other analyzer action during its compilation start action.
-                    // We need to ensure that we have force analyzed all documents in this project for this analyzer before executing the end actions.
-                    forceAnalyzeAllDocuments(_project, analyzer, _cancellationToken);
-                }
+                    // Get all the analyzer actions, including the per-compilation actions.
+                    var analyzerExecutor = GetAnalyzerExecutor(analyzer, compilation, localDiagnostics.Add);
+                    var analyzerActions = await this.GetAnalyzerActionsAsync(analyzer, analyzerExecutor).ConfigureAwait(false);
+                    var hasDependentCompilationEndAction = await AnalyzerManager.Instance.GetAnalyzerHasDependentCompilationEndAsync(analyzer, analyzerExecutor).ConfigureAwait(false);
+
+                    if (hasDependentCompilationEndAction && forceAnalyzeAllDocuments != null)
+                    {
+                        // Analyzer registered a compilation end action and at least one other analyzer action during its compilation start action.
+                        // We need to ensure that we have force analyzed all documents in this project for this analyzer before executing the end actions.
+                        // Doing so on the original analyzer instance might cause duplicate callbacks into the analyzer.
+                        // So we create a new instance of this analyzer and compute compilation diagnostics on that instance.
+
+                        try
+                        {
+                            analyzer = Activator.CreateInstance(analyzer.GetType()) as DiagnosticAnalyzer;
+                        }
+                        catch
+                        {
+                            // Unable to created a new analyzer instance, bail out on reporting diagnostics.
+                            return;
+                        }
+
+                        analyzerExecutor = GetAnalyzerExecutor(analyzer, compilation, localDiagnostics.Add);
+                        analyzerActions = await this.GetAnalyzerActionsAsync(analyzer, analyzerExecutor).ConfigureAwait(false);
+                        analyzerToDispose = analyzer;
```

Comment:
this might leak. since analyzer will hand out to outer system. assumption was analyzer either leave forever or all handed out analyzer will be cleaned up when reference is removed. which was okay since analyzer were shared.

now, newly created analyzer is handed out. for host analyzer, newly created analyzer might live very long time.

what is the reason we need to clone? analyzer itself is stateless right? the initialize method is the starting point of the state, why not just re-initialize those?